src/AutoPilotPlugins/APM/APMSensorsComponent.qml

```diff
@@ -499,6 +522,76 @@ SetupPage {
+                                            qsTr("is difficult. Point the vehicle North before using it.")
+                            }
+
+                            Column {
+                                visible:            northCalibrationLabel.visible
+                                anchors.margins:    ScreenTools.defaultFontPixelWidth
+                                anchors.left:       parent.left
+                                anchors.right:      parent.right
+                                spacing:            ScreenTools.defaultFontPixelHeight
+
+                                QGCCheckBox {
+                                    id:             northCalibrationCheckBox
+                                    visible:        northCalibrationLabel.visible
+                                    text:           qsTr("Fast Calibration")
+                                }
+
+                                QGCLabel {
+                                    id:         northCalibrationManualPosition
+                                    width:      parent.width
+                                    visible:    northCalibrationCheckBox.checked && !globals.activeVehicle.coordinate.isValid
+                                    wrapMode:   Text.WordWrap
+                                    text:       qsTr("Vehicle has no Valid positon, please provide it")
+                                }
+
+                                QGCCheckBox {
+                                    visible:    northCalibrationManualPosition.visible && _gcsPosition.isValid
+                                    id:         useGcsPositionCheckbox
+                                    text:       qsTr("Use GCS position instead")
+                                    checked:    _gcsPosition.isValid
+                                }
+                                QGCCheckBox {
+                                    visible:    northCalibrationManualPosition.visible && !_gcsPosition.isValid
+                                    id:         useMapPositionCheckbox
+                                    text:       qsTr("Use current map position instead")
+                                }
+
+                                QGCLabel {
+                                    width:      parent.width
+                                    visible:    useMapPositionCheckbox.checked
+                                    wrapMode:   Text.WordWrap
+                                    text:       qsTr(`Lat: ${_mapPosition.latitude.toFixed(4)} Lon: ${_mapPosition.longitude.toFixed(4)}`)
+                                }
+
+                                FactTextField {
+                                    id:         northCalLat
+                                    visible:    !useGcsPositionCheckbox.checked && !useMapPositionCheckbox.checked && northCalibrationCheckBox.checked
+                                    text:       "0.00"
```

Comment:
Why _gcsPosition is not being used for latitude ?

Reply:
changed both to zero. mid-writing this I decided to use a label when showing the gcs position, instead of the textfield.